src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,213 @@
+    auto expectedNBins = (*expectedMaximum - *expectedMinimum) + 1;
+
+    // Set up data
+    Data1D data;
+    data.initialise(expectedNBins, true);
+    auto x = *expectedMinimum;
+    for (auto n = 0; n < expectedNBins; ++n)
+        data.xAxis(n) = x++;
+
+    return std::make_pair(data, expectedMinimum);
+}
+
+// Initialise with specified bin range
+void IntegerHistogram1D::initialise(std::optional<int> xMin, std::optional<int> xMax)
+{
+    clear();
+
+    minimum_ = xMin;
+    maximum_ = xMax;
+
+    raw_.clear();
+    averages_.clear();
+}
+
+// Zero histogram bins
+void IntegerHistogram1D::zeroBins()
+{
+    raw_.clear();
+    averages_.clear();
+    nBinned_ = 0;
+    nMissed_ = 0;
+}
+
+// Return minimum value for data
+std::optional<int> IntegerHistogram1D::minimum() const { return minimum_; }
+
+// Return maximum value for data
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    if ((minimum_ && x < *minimum_) || (maximum_ && x > *maximum_))
+    {
+        nMissed_++;
+        return false;
+    }
+
+    raw_[x]++;
+    ++nBinned_;
+
+    return true;
+}
+
+// Return number of values binned over all bins
+long int IntegerHistogram1D::nBinned() const { return nBinned_; }
+
+// Accumulate current histogram bins into averages
+void IntegerHistogram1D::accumulate()
+{
+    for (auto &[key, value] : raw_)
+        averages_[key] += (SampledDouble)raw_[key];
+
+    // Update accumulated data
+    updateAccumulatedData();
+}
+
+// Return current data
+Data1D IntegerHistogram1D::data() const
+{
+    Data1D result = accumulatedData_;
+    for (auto &[key, value] : raw_)
+    {
+        result.values()[(key)] = value;
+    }
+    return result;
+}
+
+// Return accumulated (averaged) data
+const Data1D &IntegerHistogram1D::accumulatedData() const { return accumulatedData_; }
+
+/*
+ * Serialisation
+ */
+
+// Read data through specified LineParser
+bool IntegerHistogram1D::deserialise(LineParser &parser)
+{
+    clear();
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    std::optional<int> minLim = parser.argb(0) ? std::optional<int>(parser.argi(1)) : std::nullopt;
+    std::optional<int> maxLim = parser.argb(2) ? std::optional<int>(parser.argi(3)) : std::nullopt;
+    initialise(minLim, maxLim);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    nBinned_ = parser.argli(0);
+    nMissed_ = parser.argli(1);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+
+    auto nBins = parser.argli(0);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    for (int n = 0; n <= nBins; ++n)
+    {
+        auto bin = parser.argi(0);
+        raw_[bin] = parser.argli(1);
+        averages_[bin].deserialise(parser);
+    }
+    return true;
```

Comment:
We don't cover this (de)serialise yet with a system test - we will when the Histogram module comes along - but this definitely wouldn't have been happy!
```suggestion
    for (auto n = 0; n < nBins; ++n)
    {
        if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
            return false;

        auto bin = parser.argi(0);
        raw_[bin] = parser.argli(1);
        averages_[bin].deserialise(parser);
    }
    
    return true;
```